src/main/java/net/geforcemods/securitycraft/gui/GuiSRAT.java

```diff
@@ -0,0 +1,264 @@
+		this.drawTexturedModalRect(startX, startY, 0, 0, xSize, ySize);
+	}
+
+
+	/**
+	 * Change the sentry mode, and update GUI buttons state
+	 */
+	protected void performSingleAction(int sentry, int mode){
+		int[] coords = getSentryCoordinates(sentry);
+		List<EntitySentry> sentries = Minecraft.getMinecraft().player.world.getEntitiesWithinAABB(EntitySentry.class,
+				new AxisAlignedBB(new BlockPos(coords[0], coords[1], coords[2])));
+		if (!sentries.isEmpty()) {
+
+			switch (mode) {
+			case AGGRESSIVE:
+				sentries.get(0).toggleMode(Minecraft.getMinecraft().player, AGGRESSIVE);
+				SecurityCraft.network.sendToServer(new PacketSetSentryMode(sentries.get(0).getPosition(), mode));
+				buttons[sentry][AGGRESSIVE].enabled = false;
+				buttons[sentry][CAMOUFLAGE].enabled = true;
+				buttons[sentry][IDLE].enabled = true;
+				break;
+			case CAMOUFLAGE:
+				sentries.get(0).toggleMode(Minecraft.getMinecraft().player, CAMOUFLAGE);
+				SecurityCraft.network.sendToServer(new PacketSetSentryMode(sentries.get(0).getPosition(), mode));
+				buttons[sentry][AGGRESSIVE].enabled = true;
+				buttons[sentry][CAMOUFLAGE].enabled = false;
+				buttons[sentry][IDLE].enabled = true;
+				break;
+			case IDLE:
+				sentries.get(0).toggleMode(Minecraft.getMinecraft().player, IDLE);
+				SecurityCraft.network.sendToServer(new PacketSetSentryMode(sentries.get(0).getPosition(), mode));
+				buttons[sentry][AGGRESSIVE].enabled = true;
+				buttons[sentry][CAMOUFLAGE].enabled = true;
+				buttons[sentry][IDLE].enabled = false;
+				break;
+			case UNBIND:
+				removeTagFromToolAndUpdate(srat, coords[0], coords[1], coords[2], Minecraft.getMinecraft().player);
+
+				for (int i = 0; i < 4; i++) {
+					buttons[sentry][i].enabled = false;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Forces sentry to a defined mode, overriding its associated button state/mode
+	 */
+	protected void forceMode(GuiButton button, int mode) {
+		int sentry = button.id / 4;
+		performSingleAction(sentry, mode);
+	}
+	
+	/**
+	 * Action to perform when a button is clicked
+	 */
+	@Override
+	protected void actionPerformed(GuiButton button) {
+		int sentry = button.id / 4;
+		int mode = button.id % 4;
+
+		if (sentry > 100) { //clicked on global buttons
+			for (int i = 0; i < buttonList.size() / 4; i++) {
+				forceMode(buttonList.get(i * 4), mode);
+			}
+			return;
+		}
+
+		performSingleAction(sentry, mode);
+	}
+
+	/**
+	 * @param sentry
+	 *            0 based
+	 */
+	private int[] getSentryCoordinates(int sentry) {
+		sentry++; // sentries are stored starting by sentry1 up to sentry6
+
+		if (srat.getItem() != null && srat.getItem() == SCContent.remoteAccessSentry && srat.getTagCompound() != null
+				&& srat.getTagCompound().getIntArray("sentry" + sentry) != null
+				&& srat.getTagCompound().getIntArray("sentry" + sentry).length > 0)
+			return srat.getTagCompound().getIntArray("sentry" + sentry);
+		else
+			return new int[] { 0, 0, 0 };
+	}
+
+	@SideOnly(Side.CLIENT)
```

Comment:
Remove.

Reply:
You have the same line in the GuiMRAT.java file https://github.com/Geforce132/SecurityCraft/blob/1935d4b1079775b5815198dd536b86b88e1dab66/src/main/java/net/geforcemods/securitycraft/gui/GuiMRAT.java#L190

Do you want me to get rid of it too?

Reply:
Actually, the reason the line is there is because the super class has it. But in this case, you added it to a custom method, which is discouraged. You should only ever use the SideOnly annotation if the vanilla overridden method/super class has it as well.

Reply:
I'm sorry, but I don't see any difference in the use of the line between the GuiMRAT and GuiSRAT files. Not trying to argue at all, just to understand the issue here. The method is the same in both classes, and the super class is too, unless I'm missing something (I won't claim I'm an expert tbh)

Reply:
Right, I should read carefully before I write. My explanation still stands, but I misread where `@SideOnly(Side.CLIENT)` is in `GuiMRAT`. You can remove both.

Reply:
I do appreciate the explanation, it was just that my head exploded trying to find the differences :) I will modify the MRAT-related files in a different Pull Request to keep issues separately